web_generator/lib/src/ast/types.dart

```diff
@@ -517,12 +518,16 @@ class _UnionDeclaration extends NamedDeclaration
       this.types = const [],
       this.isNullable = false,
       List<GenericType>? typeParameters})
-      : typeParameters = typeParameters ??
-            types.map(getGenericTypes).fold(<GenericType>[],
-                (prev, combine) => [...prev, ...combine]).map((t) {
+      : typeParameters = typeParameters ?? [] {
```

Comment:
Note that you're initializing `typeParameters` and this code will execute first so `typeParameters` will never be null, and we'll never run the body of the `if` condition. Do you mean to check if it's empty instead?